File: codeforces/71A/c/71a.c
#include <stdio.h>
#include <string.h>

#define MAX_LEN         128
#define LONG_WORD_LEN   10

int main()
{
        char word[MAX_LEN];
        int n, len;
        scanf("%d", &n);

        while (n) {
                scanf("%s", word);
                len = strlen(word);

                if (len > LONG_WORD_LEN) {
                        printf("%c%d%c\n", word[0], len-2, word[len-1]);
                }
                else {
                        printf("%s\n", word);
                }

                n--;
        }
}

File: codeforces/71A/go/71a.go
package main

import (
	"fmt"
)

const LONG_WORD_LEN int = 10

func main() {
        var n uint32
        fmt.Scanf("%d", &n)

        for n != 0 {
		var str string
		fmt.Scan(&str)

		length := len(str)
                if (length <= LONG_WORD_LEN) {
                        fmt.Printf("%s\n", str)
                } else {
                        fmt.Printf("%c%d%c\n", str[0], length - 2, str[length-1])
                }

                n--
        }
}
